Weekly `cargo update` + update allowed dependencies list

clubby789 opened the pull request:
Fixes https://github.com/rust-lang/rust/pull/145516

rust-log-analyzer commented:
The job **`x86_64-gnu-tools`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/17211576295/job/48824568831) [(plain enhanced)](https://triage.rust-lang.org/gha-logs/rust-lang/rust/48824568831) [(plain)](https://github.com/rust-lang/rust/commit/2473f30a5c07d0f71f9e4daf701abda34ad72300/checks/48824568831/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
#15 sha256:c668392adc03e53ac648471cf66c21564a6d311e0fdd72b692efdbc8ab4dedeb 369B / 369B 0.2s done
#15 sha256:a3be5d4ce40198dc77f17780f02720f55b1898a2368f701dd1619fc9f84aac86 29.54MB / 29.54MB 0.3s done
#15 sha256:be77313de1e748c58b671c36ffcc17ff7396fc09e6ec71fa40ccade7df6a4bfe 591B / 591B 0.2s done
#15 sha256:18e05f7d510829d15556c6fe0c811d1598a46f50ada7cc9878f5058586071b1e 178.94MB / 178.94MB 2.9s done
#15 0.614 error: failed to copy: httpReadSeeker: failed open: failed to do request: Get "https://pkg-containers.githubusercontent.com/ghcr1/blobs/sha256:be77313de1e748c58b671c36ffcc17ff7396fc09e6ec71fa40ccade7df6a4bfe?se=2025-08-25T14%3A35%3A00Z&sig=TIGKQfiXZraStDmMzoV26WUgsLzYqZ18uIZNCZt68qc%3D&ske=2025-08-26T12%3A16%3A03Z&skoid=fb3d2a07-ec6c-4fe4-aced-9efe0fd2fe1a&sks=b&skt=2025-08-25T12%3A16%3A03Z&sktid=398a6654-997b-47e9-b12b-9515b896b4de&skv=2025-01-05&sp=r&spr=https&sr=b&sv=2025-01-05&hmac=7df1a71c4afc256c0b71b1e9aae7658903f9d7a7ecdfe5d1b11e814a86fc64b4": read tcp 172.17.0.2:38608->185.199.109.154:443: read: connection reset by peer
#15 0.614 retrying in 1s
#15 DONE 3.5s

#14 exporting to docker image format
#14 sending tarball
---

warning: `test_docs` (lib doc) generated 4 warnings
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 1.54s
   Generated /checkout/obj/build/x86_64-unknown-linux-gnu/test/rustdoc-gui/doc/test_docs/index.html
npm ERR! code E403
npm ERR! 403 403 Forbidden - GET https://registry.npmjs.org/zod/-/zod-3.23.8.tgz
npm ERR! 403 In most cases, you or one of your dependencies are requesting
npm ERR! 403 a package version that is forbidden by your security policy, or
npm ERR! 403 on a server you do not have access to.

npm ERR! A complete log of this run can be found in: /home/user/.npm/_logs/2025-08-25T15_29_41_559Z-debug-0.log
npm install did not exit successfully

thread 'main' panicked at src/tools/rustdoc-gui-test/src/main.rs:69:10:
unable to install browser-ui-test: Custom { kind: Other, error: "npm install returned exit code exit status: 1" }
stack backtrace:
   0: __rustc::rust_begin_unwind
             at /rustc/788da80fcfcef3f34c90def5baa32813e39a1a41/library/std/src/panicking.rs:697:5
   1: core::panicking::panic_fmt
             at /rustc/788da80fcfcef3f34c90def5baa32813e39a1a41/library/core/src/panicking.rs:75:14
```

</details>

samueltardieu commented:
Error looks spurious, restarting the jobs to check.

samueltardieu commented:
@clubby789 #145858 is merged, you may rebase

rust-log-analyzer commented:
The job **`aarch64-gnu-llvm-19-1`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/17252644191/job/48958164007) [(plain enhanced)](https://triage.rust-lang.org/gha-logs/rust-lang/rust/48958164007) [(plain)](https://github.com/rust-lang/rust/commit/300b3f7b7636f313d7e1f300caf146c02a03f27e/checks/48958164007/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
##[endgroup]
Doc path: /checkout/obj/build/aarch64-unknown-linux-gnu/doc/alloc/index.html
[TIMING:end] doc::Std { build_compiler: Compiler { stage: 2, host: aarch64-unknown-linux-gnu, forced_compiler: false }, target: aarch64-unknown-linux-gnu, format: Html, crates: [] } -- 53.898
##[group]Testing stage3 rustdoc-js-std (stage2 -> stage3, aarch64-unknown-linux-gnu)
Error: mismatch 124928 75,0,0,1,0,19,0,2,181,208,62,69,0,1,32,110,0,1,2,0,243,198,60,0,0,1,0,1,0,0,0,1,0,208,0,5,63,0,1,61,242,81,0,0,0,1,0,0,0,0,0,0,243,60,5,0,0,1,0,1,0,0,0,2,63,125,0,0,243,198,60,0,0,1,0,1,0,0,242,0,2,0,58,125,255,1,3,0,243,140,105,0,0,1,0,158,34,1,3,0,244,136,105,0,0,1,0,1,0,1,0,243,0,2,99,108,1,47,57,0,0,1,0,0,244,64,111,0,0,247,28,48,25,183,22,1,1,0,246,49,46,1,0,1,0,232,1,1,0,59,0,1,0,1,1,0,246,47,46,1,0,1
    at new SearchTreeBranchesAlphaBitmap (/checkout/obj/build/aarch64-unknown-linux-gnu/doc/static.files/stringdex-0e748618.js:1:22251)
    at makeBranchesFromBinaryData (/checkout/obj/build/aarch64-unknown-linux-gnu/doc/static.files/stringdex-0e748618.js:2:8210)
    at makeSearchTreeFromBase64 (/checkout/obj/build/aarch64-unknown-linux-gnu/doc/static.files/stringdex-0e748618.js:2:12462)
    at rr_ (/checkout/obj/build/aarch64-unknown-linux-gnu/doc/static.files/stringdex-0e748618.js:1:16610)
    at eval (eval at loadRoot (/checkout/src/tools/rustdoc-js/tester.js:461:17), <anonymous>:1:1)
    at Object.loadRoot (/checkout/src/tools/rustdoc-js/tester.js:461:17)
    at /checkout/obj/build/aarch64-unknown-linux-gnu/doc/static.files/stringdex-0e748618.js:2:14195
    at new Promise (<anonymous>)
    at Object.loadDatabase (/checkout/obj/build/aarch64-unknown-linux-gnu/doc/static.files/stringdex-0e748618.js:2:14065)
    at Object.initSearch (/checkout/obj/build/aarch64-unknown-linux-gnu/doc/static.files/search-5fc2c5c0.js:5:11634)
Command `/usr/bin/node /checkout/src/tools/rustdoc-js/tester.js --crate-name std --resource-suffix 1.91.0 --doc-folder /checkout/obj/build/aarch64-unknown-linux-gnu/doc --test-folder /checkout/tests/rustdoc-js-std` failed with exit code 1
Created at: src/bootstrap/src/core/build_steps/test.rs:957:27
Executed at: src/bootstrap/src/core/build_steps/test.rs:990:17

Command has failed. Rerun with -v to see more details.
Bootstrap failed while executing `--stage 2 test --skip compiler --skip src`
Build completed unsuccessfully in 0:51:28
  local time: Wed Aug 27 00:06:40 UTC 2025
  network time: Wed, 27 Aug 2025 00:06:40 GMT
##[error]Process completed with exit code 1.
```

</details>

klensy commented:
Oh, yes, stringdex is tested to recover perf in #145911, so shouldn't be bumped.

clubby789 commented:
Looks like one of the bumps (`rustfix`?) has improved and removed some duplication in `.fixed` files

bors commented:
:umbrella: The latest upstream changes (presumably #146700) made this pull request unmergeable. Please [resolve the merge conflicts](https://rustc-dev-guide.rust-lang.org/git.html#rebasing-and-conflicts).

Mark-Simulacrum commented on src/tools/tidy/src/deps.rs:
```diff
@@ -438,7 +436,7 @@ const PERMITTED_RUSTC_DEPENDENCIES: &[&str] = &[
     "windows_x86_64_gnullvm",
     "windows_x86_64_msvc",
-    "wit-bindgen-rt@0.39.0", // pinned to a specific version due to using a binary blob: <https://github.com/rust-lang/rust/pull/136395#issuecomment-2692769062>
+    "wit-bindgen",
```
I think no, https://docs.rs/crate/wit-bindgen/latest/source/build.rs looks like it's pulling in binary blobs from https://docs.rs/crate/wit-bindgen/latest/source/src/rt/, so it should remain pinned (just we now need to pin a different crate).

klensy commented on Cargo.lock:
```diff
@@ -5241,9 +5237,9 @@ dependencies = [
 [[package]]
 name = "stringdex"
-version = "0.0.1-alpha10"
+version = "0.0.1-alpha9"
```
bad rebase?

clubby789 replied:
I think lexical sorting is leading to alpha10 being treated as the version between alpha1 and alpha2, so `cargo update` pulls in the 'newest' version, alpha9
<img width="585" height="178" alt="image" src="https://github.com/user-attachments/assets/1df41f79-56b1-4238-92ff-45c1b9fb4048" />
<img width="155" height="145" alt="image" src="https://github.com/user-attachments/assets/d1456bdd-f79e-46fe-9137-96d47d8a08dc" />

notriddle replied:
Ouch, good point.

The `=` version is fine for now. I'll switch to 0.0.2 next time I push an update.

bors commented:
:umbrella: The latest upstream changes (presumably #147405) made this pull request unmergeable. Please [resolve the merge conflicts](https://rustc-dev-guide.rust-lang.org/git.html#rebasing-and-conflicts).

clubby789 committed:
Run `cargo update`

compiler dependencies:
    Updating addr2line v0.24.2 -> v0.25.1
    Removing android-tzdata v0.1.1
    Updating anstream v0.6.20 -> v0.6.21
    Updating anstyle v1.0.11 -> v1.0.13
    Updating anyhow v1.0.99 -> v1.0.100
    Updating backtrace v0.3.75 -> v0.3.76
    Updating bitflags v2.9.3 -> v2.9.4
    Updating camino v1.1.11 -> v1.2.1
    Updating cfg-if v1.0.1 -> v1.0.3
    Updating chrono v0.4.41 -> v0.4.42
    Updating clap v4.5.45 -> v4.5.48
    Updating clap_builder v4.5.44 -> v4.5.48
    Updating clap_derive v4.5.45 -> v4.5.47
    Updating ctrlc v3.4.7 -> v3.5.0
    Updating cxx v1.0.185 -> v1.0.186
    Updating cxx-build v1.0.185 -> v1.0.186
    Updating cxxbridge-cmd v1.0.185 -> v1.0.186
    Updating cxxbridge-flags v1.0.185 -> v1.0.186
    Updating cxxbridge-macro v1.0.185 -> v1.0.186
    Updating dbus v0.9.7 -> v0.9.9
      Adding dispatch v0.2.0
    Updating dyn-clone v1.0.19 -> v1.0.20
    Updating erased-serde v0.4.6 -> v0.4.8
    Updating errno v0.3.13 -> v0.3.14
    Updating filetime v0.2.25 -> v0.2.26
    Updating find-msvc-tools v0.1.2 -> v0.1.3
    Updating flate2 v1.1.2 -> v1.1.4
    Updating form_urlencoded v1.2.1 -> v1.2.2
    Updating gimli v0.32.0 -> v0.32.3
      Adding hashbrown v0.16.0
    Updating humantime v2.2.0 -> v2.3.0
    Updating iana-time-zone v0.1.63 -> v0.1.64
    Updating idna v1.0.3 -> v1.1.0
    Updating ipc-channel v0.20.1 -> v0.20.2
    Updating jobserver v0.1.33 -> v0.1.34
    Updating js-sys v0.3.77 -> v0.3.81
    Updating libdbus-sys v0.2.5 -> v0.2.6
    Updating libffi v4.1.1 -> v4.1.2
    Updating libffi-sys v3.3.2 -> v3.3.3
    Updating libloading v0.8.8 -> v0.8.9
    Updating libredox v0.1.9 -> v0.1.10
    Updating linux-raw-sys v0.9.4 -> v0.11.0
    Updating lock_api v0.4.13 -> v0.4.14
    Updating log v0.4.27 -> v0.4.28
    Updating matchers v0.1.0 -> v0.2.0
    Updating memchr v2.7.5 -> v2.7.6
    Updating normpath v1.3.0 -> v1.5.0
    Removing nu-ansi-term v0.46.0
    Updating objc2-core-foundation v0.3.1 -> v0.3.2
    Updating objc2-io-kit v0.3.1 -> v0.3.2
    Removing overload v0.1.1
    Updating owo-colors v4.2.2 -> v4.2.3
    Updating parking_lot v0.12.4 -> v0.12.5
    Updating parking_lot_core v0.9.11 -> v0.9.12
    Updating percent-encoding v2.3.1 -> v2.3.2
    Updating pest v2.8.1 -> v2.8.3
    Updating pest_derive v2.8.1 -> v2.8.3
    Updating pest_generator v2.8.1 -> v2.8.3
    Updating pest_meta v2.8.1 -> v2.8.3
    Updating potential_utf v0.1.2 -> v0.1.3
    Updating quote v1.0.40 -> v1.0.41
    Updating redox_syscall v0.5.17 -> v0.5.18
    Updating ref-cast v1.0.24 -> v1.0.25
    Updating ref-cast-impl v1.0.24 -> v1.0.25
    Updating regex v1.11.1 -> v1.11.3
    Removing regex-automata v0.1.10
    Removing regex-automata v0.4.9
      Adding regex-automata v0.4.11
    Updating regex-lite v0.1.6 -> v0.1.7
    Removing regex-syntax v0.6.29
    Removing regex-syntax v0.8.5
      Adding regex-syntax v0.8.6
    Updating rustix v1.0.8 -> v1.1.2
    Updating schannel v0.1.27 -> v0.1.28
    Updating semver v1.0.26 -> v1.0.27
    Updating serde-untagged v0.1.8 -> v0.1.9
    Updating serde_json v1.0.142 -> v1.0.145
    Updating serde_path_to_error v0.1.17 -> v0.1.20
      Adding simd-adler32 v0.3.7
    Updating sysinfo v0.37.0 -> v0.37.2
    Updating tempfile v3.20.0 -> v3.23.0
    Updating thiserror v2.0.15 -> v2.0.17
    Updating thiserror-impl v2.0.15 -> v2.0.17
    Updating tinyvec v1.9.0 -> v1.10.0
    Updating tracing-subscriber v0.3.18 -> v0.3.20
    Updating twox-hash v2.1.1 -> v2.1.2
    Updating typenum v1.18.0 -> v1.19.0
    Updating unicode-ident v1.0.18 -> v1.0.19
    Updating unicode-width v0.2.1 -> v0.2.2
    Updating url v2.5.4 -> v2.5.7
    Updating uuid v1.18.0 -> v1.18.1
    Updating wasi v0.14.2+wasi-0.2.4 -> v0.14.7+wasi-0.2.4
    Updating wasi-preview1-component-adapter-provider v36.0.1 -> v36.0.2
      Adding wasip2 v1.0.1+wasi-0.2.4
    Updating wasm-bindgen v0.2.100 -> v0.2.104
    Updating wasm-bindgen-backend v0.2.100 -> v0.2.104
    Updating wasm-bindgen-macro v0.2.100 -> v0.2.104
    Updating wasm-bindgen-macro-support v0.2.100 -> v0.2.104
    Updating wasm-bindgen-shared v0.2.100 -> v0.2.104
    Updating winapi-util v0.1.9 -> v0.1.11
    Removing windows v0.58.0
    Updating windows-core v0.58.0 -> v0.62.2
    Removing windows-implement v0.58.0
    Removing windows-implement v0.60.0
      Adding windows-implement v0.60.2
    Removing windows-interface v0.58.0
    Removing windows-interface v0.59.1
      Adding windows-interface v0.59.3
      Adding windows-link v0.2.1
    Updating windows-result v0.2.0 -> v0.4.1
    Updating windows-strings v0.1.0 -> v0.5.1
      Adding windows-sys v0.61.2
    Updating windows-targets v0.53.3 -> v0.53.5
    Updating windows_aarch64_gnullvm v0.53.0 -> v0.53.1
    Updating windows_aarch64_msvc v0.53.0 -> v0.53.1
    Updating windows_i686_gnu v0.53.0 -> v0.53.1
    Updating windows_i686_gnullvm v0.53.0 -> v0.53.1
    Updating windows_i686_msvc v0.53.0 -> v0.53.1
    Updating windows_x86_64_gnu v0.53.0 -> v0.53.1
    Updating windows_x86_64_gnullvm v0.53.0 -> v0.53.1
    Updating windows_x86_64_msvc v0.53.0 -> v0.53.1
      Adding wit-bindgen v0.46.0
    Removing wit-bindgen-rt v0.39.0
    Updating xattr v1.5.1 -> v1.6.1
    Updating zerocopy v0.8.26 -> v0.8.27
    Updating zerocopy-derive v0.8.26 -> v0.8.27

library dependencies:
    Updating addr2line v0.25.0 -> v0.25.1
    Updating cfg-if v1.0.1 -> v1.0.3
    Updating dlmalloc v0.2.10 -> v0.2.11
    Updating gimli v0.32.0 -> v0.32.3
    Updating libc v0.2.175 -> v0.2.176
    Updating memchr v2.7.5 -> v2.7.6
    Updating vex-sdk v0.27.0 -> v0.27.1
      Adding windows-link v0.2.1
    Updating windows-sys v0.59.0 -> v0.60.2
    Updating windows-targets v0.52.6 -> v0.53.5
    Updating windows_aarch64_gnullvm v0.52.6 -> v0.53.1
    Updating windows_aarch64_msvc v0.52.6 -> v0.53.1
    Updating windows_i686_gnu v0.52.6 -> v0.53.1
    Updating windows_i686_gnullvm v0.52.6 -> v0.53.1
    Updating windows_i686_msvc v0.52.6 -> v0.53.1
    Updating windows_x86_64_gnu v0.52.6 -> v0.53.1
    Updating windows_x86_64_gnullvm v0.52.6 -> v0.53.1
    Updating windows_x86_64_msvc v0.52.6 -> v0.53.1
    Updating wit-bindgen v0.45.0 -> v0.45.1

rustbook dependencies:
    Updating crates.io index
     Locking 82 packages to latest compatible versions
    Updating ammonia v4.1.1 -> v4.1.2
    Removing android-tzdata v0.1.1
    Updating anstream v0.6.20 -> v0.6.21
    Updating anstyle v1.0.11 -> v1.0.13
    Updating anyhow v1.0.99 -> v1.0.100
    Removing bitflags v1.3.2
    Removing bitflags v2.9.2
      Adding bitflags v2.9.4
    Updating cc v1.2.33 -> v1.2.40
    Updating cfg-if v1.0.1 -> v1.0.3
    Updating chrono v0.4.41 -> v0.4.42
    Updating clap v4.5.45 -> v4.5.48
    Updating clap_builder v4.5.44 -> v4.5.48
    Updating clap_complete v4.5.57 -> v4.5.58
    Updating clap_derive v4.5.45 -> v4.5.47
    Updating errno v0.3.13 -> v0.3.14
      Adding find-msvc-tools v0.1.3
    Updating flate2 v1.1.2 -> v1.1.4
    Updating form_urlencoded v1.2.1 -> v1.2.2
    Updating getopts v0.2.23 -> v0.2.24
    Updating hashbrown v0.15.5 -> v0.16.0
    Updating iana-time-zone v0.1.63 -> v0.1.64
    Updating idna v1.0.3 -> v1.1.0
    Updating indexmap v2.10.0 -> v2.11.4
    Updating js-sys v0.3.77 -> v0.3.81
    Updating linux-raw-sys v0.9.4 -> v0.11.0
    Updating lock_api v0.4.13 -> v0.4.14
    Updating log v0.4.27 -> v0.4.28
    Updating memchr v2.7.5 -> v2.7.6
    Updating normpath v1.3.0 -> v1.5.0
    Updating opener v0.8.2 -> v0.8.3
    Updating parking_lot v0.12.4 -> v0.12.5
    Updating parking_lot_core v0.9.11 -> v0.9.12
    Updating percent-encoding v2.3.1 -> v2.3.2
    Updating pest v2.8.1 -> v2.8.3
    Updating pest_derive v2.8.1 -> v2.8.3
    Updating pest_generator v2.8.1 -> v2.8.3
    Updating pest_meta v2.8.1 -> v2.8.3
    Updating potential_utf v0.1.2 -> v0.1.3
    Updating quote v1.0.40 -> v1.0.41
    Updating redox_syscall v0.5.17 -> v0.5.18
    Updating regex v1.11.1 -> v1.11.3
    Updating regex-automata v0.4.9 -> v0.4.11
    Updating regex-syntax v0.8.5 -> v0.8.6
    Updating rustix v1.0.8 -> v1.1.2
    Updating semver v1.0.26 -> v1.0.27
    Updating serde v1.0.219 -> v1.0.228
      Adding serde_core v1.0.228
    Updating serde_derive v1.0.219 -> v1.0.228
    Updating serde_json v1.0.142 -> v1.0.145
      Adding simd-adler32 v0.3.7
    Updating syntect v5.2.0 -> v5.3.0
    Updating tempfile v3.20.0 -> v3.23.0
    Updating thiserror v2.0.15 -> v2.0.17
    Updating thiserror-impl v2.0.15 -> v2.0.17
    Updating typenum v1.18.0 -> v1.19.0
    Updating unicode-ident v1.0.18 -> v1.0.19
    Updating unicode-width v0.2.1 -> v0.2.2
    Updating url v2.5.4 -> v2.5.7
    Updating wasi v0.14.2+wasi-0.2.4 -> v0.14.7+wasi-0.2.4
      Adding wasip2 v1.0.1+wasi-0.2.4
    Updating wasm-bindgen v0.2.100 -> v0.2.104
    Updating wasm-bindgen-backend v0.2.100 -> v0.2.104
    Updating wasm-bindgen-macro v0.2.100 -> v0.2.104
    Updating wasm-bindgen-macro-support v0.2.100 -> v0.2.104
    Updating wasm-bindgen-shared v0.2.100 -> v0.2.104
    Updating winapi-util v0.1.9 -> v0.1.11
    Updating windows-core v0.61.2 -> v0.62.2
    Updating windows-implement v0.60.0 -> v0.60.2
    Updating windows-interface v0.59.1 -> v0.59.3
    Updating windows-link v0.1.3 -> v0.2.1
    Updating windows-result v0.3.4 -> v0.4.1
    Updating windows-strings v0.4.2 -> v0.5.1
    Updating windows-sys v0.59.0 -> v0.61.2
    Removing windows-targets v0.52.6
    Removing windows-targets v0.53.3
      Adding windows-targets v0.53.5
    Removing windows_aarch64_gnullvm v0.52.6
    Removing windows_aarch64_gnullvm v0.53.0
      Adding windows_aarch64_gnullvm v0.53.1
    Removing windows_aarch64_msvc v0.52.6
    Removing windows_aarch64_msvc v0.53.0
      Adding windows_aarch64_msvc v0.53.1
    Removing windows_i686_gnu v0.52.6
    Removing windows_i686_gnu v0.53.0
      Adding windows_i686_gnu v0.53.1
    Removing windows_i686_gnullvm v0.52.6
    Removing windows_i686_gnullvm v0.53.0
      Adding windows_i686_gnullvm v0.53.1
    Removing windows_i686_msvc v0.52.6
    Removing windows_i686_msvc v0.53.0
      Adding windows_i686_msvc v0.53.1
    Removing windows_x86_64_gnu v0.52.6
    Removing windows_x86_64_gnu v0.53.0
      Adding windows_x86_64_gnu v0.53.1
    Removing windows_x86_64_gnullvm v0.52.6
    Removing windows_x86_64_gnullvm v0.53.0
      Adding windows_x86_64_gnullvm v0.53.1
    Removing windows_x86_64_msvc v0.52.6
    Removing windows_x86_64_msvc v0.53.0
      Adding windows_x86_64_msvc v0.53.1
    Updating winnow v0.7.12 -> v0.7.13
      Adding wit-bindgen v0.46.0
    Removing wit-bindgen-rt v0.39.0
note: pass `--verbose` to see 1 unchanged dependencies behind latest

clubby789 committed:
Pin `stringdex`

clubby789 committed:
Update allowed dependencies list

bjorn3 commented on src/tools/tidy/src/deps.rs:
```diff
@@ -537,7 +535,7 @@ const PERMITTED_RUSTC_DEPENDENCIES: &[&str] = &[
     "windows_x86_64_gnullvm",
     "windows_x86_64_msvc",
-    "wit-bindgen-rt@0.39.0", // pinned to a specific version due to using a binary blob: <https://github.com/rust-lang/rust/pull/136395#issuecomment-2692769062>
+    "wit-bindgen@0.46.0", // pinned to a specific version due to using a binary blob: <https://github.com/rust-lang/rust/pull/136395#issuecomment-2692769062>
```
`wit-bindgen@0.46.0` contains a binary blob. It seems like we accidentally already got a dependency on this blob for the standard library again though.

alexcrichton replied:
Ah yes the binary blob there is expected, and I'm happy to provide context for it if necessary. (unsure if you were looking for any other sort of particular feedback here)

bjorn3 replied:
My understanding is that we intended to pin wit-bindgen-rt to a version from before the binary blob was introduced because some distros don't like binary blobs in source tarballs. I assume the reason you are using a binary blob is because of weak symbols. Would something like the binary blobs into a separate crate and then replacing this crate on the rustc side with `[patch.crates-io]` with a pure rust implementation that uses the unstable `#[linkage = "weak"]` or something like that work?
